### PR TITLE
feature: create Purchase UI

### PR DESCRIPTION
**Pull request by DarrylEgbertNangoi-GitHub**

### CODE REVIEW

**Review by LeroyMichael — CHANGES_REQUESTED** [review]
Request fix/add
- add wallets (Select) | get data from getWallets
- tambah tax fields (tax_rate selalu percentage) show juga total fix taxnya brapa

Notes
- pas select contact nanti backgroundnya jd gray | contoh di transactions
- add items perlu default value 1 row kosong & tambahin subtotal di atas add new item

File: app/(protected)/purchases/[...purchase]/components/PurchaseCustomerDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+                      <Textarea
+                        placeholder="e.g. Jalan Indonesia Blok C No. 22"
+                        {...field}
+                        />
+                  </FormItem>
+                )}
+              />
+            </div>
+          </div>
+          <ScrollArea className="h-[300px]">
```
Comment (LeroyMichael) [inline app/(protected)/purchases/[...purchase]/components/PurchaseCustomerDetails.tsx:75]: perlu tambah search contact + logicnya 
[reference](https://github.com/LeroyMichael/PowerUp/blob/ef6cefb4a8cb625d06a26251a1ac5919f23818b4/app/(protected)/sales/%5B...sale%5D/contactDetail.tsx#L101)
di paling bawah ada paginationnya
image

File: app/(protected)/purchases/[...purchase]/components/PurchaseSubtotal.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            <div className="flex justify-between">
+              <span>Tax</span>
+              <span>Rp{numberFixedToString(watchForm.tax)} </span>
+            </div>
+
+            <Separator/>
+
+            <div className="flex justify-between">
+              <span>Total</span>
+              <span>Rp{numberFixedToString(watchForm.total)}</span>
```
Comment (LeroyMichael) [inline app/(protected)/purchases/[...purchase]/components/PurchaseSubtotal.tsx:62]: ini numberfixedtostring ganti jadi 
`
<NumericFormat
                                value={watchForm.total}
                                displayType={"text"}
                                prefix={"Rp"}
                                allowNegative={false}
                                decimalSeparator={","}
                                thousandSeparator={"."}
                                fixedDecimalScale={true}
                              />
`

Comment (LeroyMichael) [reply app/(protected)/purchases/[...purchase]/components/PurchaseSubtotal.tsx:62]: berlaku buat semua price value ya
image

**Review by LeroyMichael — CHANGES_REQUESTED** [review]

File: app/(protected)/purchases/[...purchase]/components/PurchaseSubtotal.tsx (hunk 2 above)
Comment (DarrylEgbertNangoi-GitHub) [reply app/(protected)/purchases/[...purchase]/components/PurchaseSubtotal.tsx]: done, tapi hasilnya Rp2.000 gada ,00

File: app/(protected)/purchases/[...purchase]/components/PurchaseCustomerDetails.tsx (hunk 1 above)
Comment (DarrylEgbertNangoi-GitHub) [reply app/(protected)/purchases/[...purchase]/components/PurchaseCustomerDetails.tsx:74]: done

**Pull request merged**